### PR TITLE
Remove Viewflow from AccessRequest models

### PR DESCRIPTION
**Pull request by kevinetienne**
This PR removes Viewflow from AccessRequest. It would be merged in a feature branch where views and forms would follow.

### CODE REVIEW

File: web/domains/case/access/models.py [hunk 1]
```diff
@@ -98,112 +58,67 @@ class AccessRequest(models.Model):
...
+    link = models.ForeignKey(
+        Exporter, on_delete=models.PROTECT, blank=True, null=True, related_name="access_requests"
+    )
+    request_type = models.CharField(
+        max_length=30,
+        choices=REQUEST_TYPES,
+        blank=False,
+        null=False,
+        verbose_name="Access Request Type",
+    )
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 2]
```diff
@@ -98,112 +58,67 @@ class AccessRequest(models.Model):
...
     )
-    linked_exporter = models.ForeignKey(
-        Exporter, on_delete=models.PROTECT, blank=True, null=True, related_name="access_requests"
+    request_type = models.CharField(
+        max_length=30,
+        choices=REQUEST_TYPES,
+        blank=False,
+        null=False,
+        verbose_name="Access Request Type",
     )
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 3]
```diff
@@ -98,112 +58,67 @@ class AccessRequest(models.Model):
...
+    eori_number = models.CharField(
+        max_length=20,
+        blank=True,
+        null=True,
+        verbose_name="EORI Number",
+        help_text="EORI number should include the GB prefix for organisation or GBPR for individual",
+    )
+    eori_number_ni = models.CharField(
+        max_length=20, blank=True, null=True, verbose_name="NI EORI Number"
+    )
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 4]
```diff
@@ -98,112 +58,67 @@ class AccessRequest(models.Model):
...
+        ("MAIN_IMPORTER_ACCESS", "Request access to act as an Importer"),
+        (AGENT_ACCESS, "Request access to act as an Agent for an Importer"),
+    )
+    eori_number = models.CharField(
+        max_length=20,
+        blank=True,
+        null=True,
+        verbose_name="EORI Number",
+        help_text="EORI number should include the GB prefix for organisation or GBPR for individual",
+    )
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 5]
```diff
@@ -1,83 +1,43 @@
...
+    request_reason = models.TextField(
+        blank=False, verbose_name="What are you importing and where are you importing it from?"
+    )
     agent_name = models.CharField(max_length=100, blank=True, null=True)
-    agent_address = models.CharField(max_length=500, blank=True, null=True)
-    submit_datetime = models.DateTimeField(blank=False, null=False)
+    agent_address = models.TextField(blank=True, default="")
+
+    submit_datetime = models.DateTimeField(blank=False, null=False, auto_now=True)
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 6]
```diff
@@ -1,83 +1,43 @@
...
     organisation_name = models.CharField(max_length=100, blank=False, null=False)
-    organisation_address = models.CharField(max_length=500, blank=False, null=True)
-    request_reason = models.CharField(max_length=1000, blank=True, null=True)
+    organisation_address = models.TextField(blank=False)
+    organisation_registered_number = models.CharField(
+        max_length=100, blank=True, null=True, verbose_name="Registered Number"
+    )
+    request_reason = models.TextField(
+        blank=False, verbose_name="What are you importing and where are you importing it from?"
+    )
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 7]
```diff
@@ -1,83 +1,43 @@
...
     status = models.CharField(
         max_length=30, choices=STATUSES, blank=False, null=False, default=SUBMITTED
     )
     organisation_name = models.CharField(max_length=100, blank=False, null=False)
-    organisation_address = models.CharField(max_length=500, blank=False, null=True)
-    request_reason = models.CharField(max_length=1000, blank=True, null=True)
+    organisation_address = models.TextField(blank=False)
+    organisation_registered_number = models.CharField(
+        max_length=100, blank=True, null=True, verbose_name="Registered Number"
+    )
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/models.py [hunk 8]
```diff
@@ -1,83 +1,43 @@
...
-    objects = AccessRequestQuerySet.as_manager()
     reference = models.CharField(max_length=50, blank=False, null=False)
-    request_type = models.CharField(max_length=30, choices=REQUEST_TYPES, blank=False, null=False)
     status = models.CharField(
         max_length=30, choices=STATUSES, blank=False, null=False, default=SUBMITTED
     )
     organisation_name = models.CharField(max_length=100, blank=False, null=False)
-    organisation_address = models.CharField(max_length=500, blank=False, null=True)
-    request_reason = models.CharField(max_length=1000, blank=True, null=True)
+    organisation_address = models.TextField(blank=False)
```
[the remark this answers is not in the document]
Comment (kevinetienne) [reply web/domains/case/access/models.py]: fixed in https://github.com/uktrade/icms/pull/205/commits/74188b843dafa8bcdaf1e285d1fdf91bc731efc9

File: web/domains/case/access/forms.py [hunk 9]
```diff
@@ -25,18 +25,16 @@ def is_valid(form, data, fields):
 
 
 def is_agent_request(request_type):
-    return request_type in [AccessRequest.IMPORTER_AGENT, AccessRequest.EXPORTER_AGENT]
+    return True
```
Comment (oskusalerma) [inline web/domains/case/access/forms.py:28]: this is temporary, so needs a FIXME or TODO comment explaining what's left to do

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/domains/case/access/forms.py (hunk 9 above)
Comment (kevinetienne) [reply web/domains/case/access/forms.py:29]: added in https://github.com/uktrade/icms/pull/205/commits/ff319e7e0c52b5240abfa456a5c71742432d0047

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**